cmake/CPM.cmake

```
@@ -862,14 +862,30 @@ function(CPMAddPackage)
   )
 
   if(NOT CPM_SKIP_FETCH)
-    cpm_declare_fetch(
-      "${CPM_ARGS_NAME}" "${CPM_ARGS_VERSION}" "${PACKAGE_INFO}" "${CPM_ARGS_UNPARSED_ARGUMENTS}"
-    )
+    # CMake 3.28 added EXCLUDE and SYSTEM(3.25) to FetchContent_Declare.
+    # Calling FetchContent_MakeAvailable will than call add_subdirectory internally with the EXCLUDE
+    # and SYSTEM flags. No need for CPM to do this manually anymore.
+    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.28.0")
+      set(fetchContentDeclareExtraArgs "")
+      if(${CPM_ARGS_EXCLUDE_FROM_ALL})
+        list(APPEND fetchContentDeclareExtraArgs EXCLUDE_FROM_ALL)
+      endif()
+      if(${CPM_ARGS_SYSTEM})
+        list(APPEND fetchContentDeclareExtraArgs SYSTEM)
+      endif()
+      cpm_declare_fetch(
```

```suggestion
      if(CPM_ARGS_OPTIONS)
        foreach(OPTION ${CPM_ARGS_OPTIONS})
          cpm_parse_option("${OPTION}")
          set(${OPTION_KEY} "${OPTION_VALUE}")
        endforeach()
      endif()
      
      cpm_declare_fetch(
```

To properly set options

---

Thanks for the feedback. Tested and fixed.